src/CoreWCF.WebHttp/src/CoreWCF/UriTemplateHelpers.cs

```diff
@@ -0,0 +1,408 @@
+                }
+
+                return UriTemplatePartType.Literal;
+            }
+            else
+            {
+                if (varEndIndex < varStartIndex + 2)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new FormatException(
+                        SR.Format(SR.UTInvalidFormatSegmentOrQueryPart, part)));
+                }
+
+                if (varStartIndex > 0)
+                {
+                    return UriTemplatePartType.Compound;
+                }
+                else if ((varEndIndex < part.Length - 2) ||
+                    ((varEndIndex == part.Length - 2) && !part.EndsWith("/", StringComparison.Ordinal)))
+                {
+                    return UriTemplatePartType.Compound;
+                }
+                else
+                {
+                    return UriTemplatePartType.Variable;
+                }
+            }
+        }
+        public static bool IsWildcardPath(string path)
+        {
+            if (path.IndexOf('/') != -1)
+            {
+                return false;
+            }
+
+            return IsWildcardSegment(path, out UriTemplatePartType partType);
+        }
+
+        public static bool IsWildcardSegment(string segment, out UriTemplatePartType type)
+        {
+            type = IdentifyPartType(segment);
+            switch (type)
+            {
+                case UriTemplatePartType.Literal:
+                    return (string.Compare(segment, UriTemplate.WildcardPath, StringComparison.Ordinal) == 0);
+
+                case UriTemplatePartType.Compound:
+                    return false;
+
+                case UriTemplatePartType.Variable:
+                    return ((segment.IndexOf(UriTemplate.WildcardPath, StringComparison.Ordinal) == 1) &&
+                        !segment.EndsWith("/", StringComparison.Ordinal) &&
+                        (segment.Length > UriTemplate.WildcardPath.Length + 2));
+
+                default:
+                    Fx.Assert("Bad part type identification !");
+                    return false;
+            }
+        }
+
+        public static NameValueCollection ParseQueryString(string query)
+        {
+            // We are adjusting the parsing of UrlUtility.ParseQueryString, which identify
+            //  ?wsdl as a null key with wsdl as a value
+            NameValueCollection result = UrlUtility.ParseQueryString(query);
+            string nullKeyValuesString = result[null];
+            if (!string.IsNullOrEmpty(nullKeyValuesString))
+            {
+                result.Remove(null);
+                string[] nullKeyValues = nullKeyValuesString.Split(',');
+                for (int i = 0; i < nullKeyValues.Length; i++)
+                {
+                    result.Add(nullKeyValues[i], null);
+                }
+            }
+
+            return result;
+        }
+
+        static bool AllTemplatesAreEquivalent(IList<UriTemplate> array, int a, int b)
+        {
```

Comment:
Missing accessibility modifiers on multiple methods in this file.

Reply:
This is addressed in e95ff7c922db5d4a6ee55847b763b49619f08311.